### PR TITLE
Created helper for inline array allocation

### PR DESCRIPTION
**Pull request by OussamaSaoudi**
Created inlineAllocateIndexableObject helper function to handle 
allocation of indexable flattened or unflattened classes by calling one
of:
- inlineAllocateIndexableValueTypeObject for flattened array classes
- inlineAllocateIndexableObject for unflattened array classes with no 
  unflattened flattenables

Replaced explicit usages of the inlineAllocateIndexableValueTypeObject 
method with the new inlineAllocateIndexableObject helper.

Related to: https://github.com/eclipse/openj9/issues/9200

Signed-off-by:Oussama Saoudi <oussama.saoudi@ibm.com>

### CODE REVIEW

**Comment by fjeremic** [comment]
Hey @OussamaSaoudi, thanks for your first contribution and welcome to the project. There seems to be quite a few foreign commits in this PR. Could you squash them into a single commit in accordance to the [Commit Guidelines](https://github.com/eclipse/openj9/blob/master/CONTRIBUTING.md#commit-guidelines).

Once that is done we'll need to make the two failed checks pass. The copyright should be easy by modifying the year of the files you changed to 2020. As for the ECA, see the [Legal considerations](https://github.com/eclipse/openj9/blob/master/CONTRIBUTING.md#legal-considerations) section in the same document as above for how to sign the ECA so your contribution can be accepted.

**Comment by tajila** [comment]
The title needs to be less than 70 characters

File: runtime/oti/VMHelpers.hpp [hunk 1]
```diff
@@ -768,6 +768,36 @@ class VM_VMHelpers
...
+				setHeapOutOfMemoryError(currentThread);
+			}
+		}
+		return instance;
+	}
+
+	/**
+	 *
+	 * Perform a non-instrumentable allocation of an indexable flattened or unflattened class.
+	 *
```
Comment (tajila) [inline runtime/oti/VMHelpers.hpp:785]: Add a comment saying "unflattened class that contain the J9ClassContainsUnflattenedFlattenables flag will return NULL"

**Comment by tajila** [comment]
Commit messages are to describe details about the PR that readers in the future might need. Things like "Tested in personal builds with no errors." can just be messages posted in the PR

**Comment by tajila** [comment]
the first line of the commit message is too long " Created inlineAllocateIndexableObject helper function to handle allocation of indexable flattened or unflattened", it should be under the 70characer limit

**Comment by tajila** [comment]
What I meant was something like this:

```
Created helper for inline array allocation

Created inlineAllocateIndexableObject helper function to handle 
allocation of indexable flattened or unflattened classes by calling one
of:
- inlineAllocateIndexableValueTypeObject for flattened array classes
- inlineAllocateIndexableObject for unflattened array classes with no 
  unflattened flattenables

Related to: https://github.com/eclipse/openj9/issues/9200

Signed-off-by: Oussama Saoudi <oussama.saoudi@ibm.com>
```
Basically you want a short title (under 70 characters) that describes the PR. Then you add a body, this can have as much detail as you like title. Then close with a sign-off.

Also, it is helpful to have the contents of your commit message in the description of the PR. If you write your commit message before you open the PR, github will attempt to use the body of the commit message as the default contents for the PR description.

File: runtime/oti/VMHelpers.hpp [hunk 2]
```diff
@@ -752,7 +752,7 @@ class VM_VMHelpers
 	 * This function assumes that the stack and live values are in a valid state for GC.
 	 *
 	 * If allocation fails, a heap OutOfMemory error is set pending on the thread.
-	 *
+	 *
```
Comment (tajila) [inline runtime/oti/VMHelpers.hpp:755]: avoid whitespace changes

File: runtime/oti/VMHelpers.hpp [hunk 3]
```diff
@@ -752,7 +752,6 @@ class VM_VMHelpers
 	 * This function assumes that the stack and live values are in a valid state for GC.
 	 *
 	 * If allocation fails, a heap OutOfMemory error is set pending on the thread.
-	 *
```
Comment (tajila) [inline runtime/oti/VMHelpers.hpp:755]: undo this change completey

**Comment by tajila** [comment]
@gacholio please review these changes

**Comment by gacholio** [comment]
Jenkins test sanity,extended xlinuxval jdknext
